prometheus/src/test/scala/filodb/prometheus/parse/ParserSpec.scala

```diff
@@ -562,8 +562,14 @@ class ParserSpec extends AnyFunSpec with Matchers {
 
-      "(metric1 + metric2) * (metric3 + metric4)" -> "BinaryJoin(BinaryJoin(PeriodicSeries(RawSeries(IntervalSelector(1524855988000,1524855988000),List(ColumnFilter(__name__,Equals(metric1))),List(),Some(300000),None),1524855988000,1000000,1524855988000,None),ADD,OneToOne,PeriodicSeries(RawSeries(IntervalSelector(1524855988000,1524855988000),List(ColumnFilter(__name__,Equals(metric2))),List(),Some(300000),None),1524855988000,1000000,1524855988000,None),List(),List(),List()),MUL,OneToOne,BinaryJoin(PeriodicSeries(RawSeries(IntervalSelector(1524855988000,1524855988000),List(ColumnFilter(__name__,Equals(metric3))),List(),Some(300000),None),1524855988000,1000000,1524855988000,None),ADD,OneToOne,PeriodicSeries(RawSeries(IntervalSelector(1524855988000,1524855988000),List(ColumnFilter(__name__,Equals(metric4))),List(),Some(300000),None),1524855988000,1000000,1524855988000,None),List(),List(),List()),List(),List(),List())"
-
+      "(metric1 + metric2) * (metric3 + metric4)" -> "BinaryJoin(BinaryJoin(PeriodicSeries(RawSeries(IntervalSelector(1524855988000,1524855988000),List(ColumnFilter(__name__,Equals(metric1))),List(),Some(300000),None),1524855988000,1000000,1524855988000,None),ADD,OneToOne,PeriodicSeries(RawSeries(IntervalSelector(1524855988000,1524855988000),List(ColumnFilter(__name__,Equals(metric2))),List(),Some(300000),None),1524855988000,1000000,1524855988000,None),List(),List(),List()),MUL,OneToOne,BinaryJoin(PeriodicSeries(RawSeries(IntervalSelector(1524855988000,1524855988000),List(ColumnFilter(__name__,Equals(metric3))),List(),Some(300000),None),1524855988000,1000000,1524855988000,None),ADD,OneToOne,PeriodicSeries(RawSeries(IntervalSelector(1524855988000,1524855988000),List(ColumnFilter(__name__,Equals(metric4))),List(),Some(300000),None),1524855988000,1000000,1524855988000,None),List(),List(),List()),List(),List(),List())",
+      "foo[5m:1m]" -> "PeriodicSeries(RawSeries(IntervalSelector(1524855688000,1524855988000),List(ColumnFilter(__name__,Equals(foo))),List(),Some(300000),None),1524855688000,60000,1524855988000,None)",
+      "rate(foo[3m])[5m:1m]" -> "PeriodicSeriesWithWindowing(RawSeries(IntervalSelector(1524855688000,1524855988000),List(ColumnFilter(__name__,Equals(foo))),List(),Some(180000),None),1524855688000,60000,1524855988000,180000,Rate,false,List(),None)"
+      //"max_over_time(rate(foo[5m])[5m:1m])" -> "" //will fail to parse
+      //"max_over_time(sum(rate(foo[5m]))[5m:1m])" -> "", //will fail to parse
+      //"sum(rate(foo[5m])[5m:1m])" -> "" // fail
```

Comment:
Can we get a query such as this to pass against parser? This is a minimal test case which makes the feature usable.

`sum(rate(foo[5m])[5m:1m])`

I believe you have made `function` as a subqueryable expression. We should get it to work.

Reply:
this one does not work yet as sub is not a function but an aggregate. I think I want to do it later, for as long as I can make function work, aggregate should come fairly easy.